src/futures_map.rs

```diff
@@ -127,15 +127,22 @@ where
     /// This iterator returns futures in an arbitrary order, which may change.
     ///
     /// If downcasting a future to `T` fails it will be skipped in the iterator.
-    pub fn iter_of_type<T>(&self) -> impl Iterator<Item = (&ID, &T)>
+    pub fn iter_of_type<T>(&self) -> impl Iterator<Item = (&ID, Pin<&T>)>
     where
         T: 'static,
     {
         self.inner.iter().filter_map(|a| {
             let pin = a.inner.inner.as_ref();
-            let any = Pin::into_inner(pin) as &(dyn Any + Send);
+
+            // Safety: We are only temporarily manipulating the pointer and pinning it again further down.
+            let pointer = unsafe { Pin::into_inner_unchecked(pin) };
+            let any = pointer as &(dyn Any + Send);
             let inner = any.downcast_ref::<T>()?;
-            Some((&a.tag, inner))
+
+            // Safety: The pointer is already pinned.
```

Comment:
```suggestion
            // Safety: The pointer is already pinned, and will remain pinned for its entire lifetime,
            // because we return a `Pin<&T>`.
```